plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._dataAttributes = {
+            columns:'timestamp,file-ref,file-name-or-duration,event-id'
+        };
+        this._tmpLogLines = []
+        this._fileName = null;
+        this._fileIndex = null;
+        this._extensionContext = extensionContext;
+        fs.mkdirSync(this._extensionContext.logUri.fsPath)
+        this._performanceTxtPath = path.join(this._extensionContext.logUri.fsPath, "qljsPerformanceTrace.txt");
+        this._currentSessionMap = this._extensionContext.globalState.get('qljsFileMap');
```

Comment:
Why does qljsFileMap need to be persisted? Could we create an empty object here instead?

I worry that maintaining the qljsFileMap globalState might be costly. We have no mechanism to clean up old data from it, for example.

Reply:
There is now an empty object specific to the instance of PerformanceWriter in latest commit